packages/vite/src/node/server/send.ts

```diff
@@ -27,6 +27,7 @@ export interface SendOptions {
   cacheControl?: string
   headers?: OutgoingHttpHeaders
   map?: SourceMap | { mappings: '' } | null
+  originalContent?: string
```

Comment:
Should this be `() => string` in case it doesn't end up being used to save the file read?

Reply:
`send` function is sync and is part of the public API. Therefore `originalContent` cannot be `() => Promise<string>`. To satisfy `() => string`, we need to change `fsp.readFile` to `fs.readFileSync`. But I guess async is better for perf.

Reply:
It would be good to find a way to avoid the perf regression. In a full .ts code base, there will now be a new read for each transformed file that won't be used, no? I think at one point we could have an internal `send` function that is async (everywhere internal usage could be changed to await), but given that it may be complex, for now we could add a guard at https://github.com/vitejs/vite/pull/15135/files#diff-6d94d6934079a4f09596acc9d3f3d38ea426c6f8e98cd766567335d42679ca7cR208, with a comment? Even if it is redundant, we at least save a few reads?

Reply:
Ah, OK. I noticed I was missing `type === 'js'`.